Customizing/global/plugins/Services/Cron/CronHook/MarkCoursesWBDRelevant/classes/class.ilMarkCoursesWBDRelevantJob.php

```diff
@@ -0,0 +1,150 @@
+
+	private $gIldb;
+	private $gLog;
+	private $gLng;
+	private $gRbacadmin;
+
+	public function __construct() {
+		global $ilDB, $ilLog, $lng, $rbacadmin;
+
+		$this->gDB = $ilDB;
+		$this->gLog = $ilLog;
+		$this->gLng = $lng;
+		$this->gRbacadmin = $rbacadmin;
+		$this->settings = gevSettings::getInstance();
+	}
+
+	/**
+	 * Implementation of abstract function from ilCronJob
+	 * @return	string
+	 */
+	public function getId() {
+		return "gev_mark_courses_wbd_relevant";
+	}
+	
+	/**
+	 * Implementation of abstract function from ilCronJob
+	 * @return	string
+	 */
+	public function getTitle() {
+		return "Setzt Trainings auf WBD relevant.";
+	}
+
+	public function getDescription() {
+		return "Wenn beim Benutzer das UDF Feld 'WBD Punkte nachmelden ab' einen Wert besitzt, werden alle Trainings deren Enddatum >= ist auf WBD relevant gestzt";
+	}
+
+	/**
+	 * Implementation of abstract function from ilCronJob
+	 * @return	bool
+	 */
+	public function hasAutoActivation() {
+		return true;
+	}
+	
+	/**
+	 * Implementation of abstract function from ilCronJob
+	 * @return	bool
+	 */
+	public function hasFlexibleSchedule() {
+		return false;
+	}
+	
+	/**
+	 * Implementation of abstract function from ilCronJob
+	 * @return	int
+	 */
+	public function getDefaultScheduleType() {
+		return ilCronJob::SCHEDULE_TYPE_DAILY;
+	}
+	
+	/**
+	 * Implementation of abstract function from ilCronJob
+	 * @return	int
+	 */
+	public function getDefaultScheduleValue() {
+		return 1;
+	}
+
+	/**
+	 * Implementation of abstract function from ilCronJob
+	 * @return	ilCronJobResult
+	 */
+	public function run() {
+		$cron_result = new ilCronJobResult();
+		$this->gLog->write("### ilMarkCoursesWBDRelevantJob: STARTING ###");
+		$user_vals = $this->getUserWithEnteredDate();
+
+		if(count($user_vals) == 0) {
+			$this->gLog->write("### ilMarkCoursesWBDRelevantJob: no user found ###");
+		}
+
+		foreach ($user_vals as $user_val) {
+			ilCronManager::ping($this->getId());
+			$this->gLog->write("### ilMarkCoursesWBDRelevantJob: set for: $user_val ###");
+			$this->updateTrainings($user_val);
+			ilCronManager::ping($this->getId());
+		}
+
+		$this->gLog->write("### ilMarkCoursesWBDRelevantJob: FINISHED ###");
+
+		$cron_result->setStatus(ilCronJobResult::STATUS_OK);
+		return $cron_result;
+	}
+
+	/**
+	 * Get all user_ids where udf field has been a value
+	 * @return [int] $ret
+	 */
+	private function getUserWithEnteredDate() {
+		$ret = array();
+		$field_id = $this->settings->getUDFFieldId(gevWBD::USR_WBD_REPORT_POINTS_FROM);
+
+		$select = "SELECT ud.usr_id, udt.value\n"
+				." FROM usr_data ud\n"
+				." JOIN udf_text udt ON udt.usr_id = ud.usr_id\n"
+				." WHERE udt.value IS NOT NULL\n"
+				."    AND udt.field_id = ".$this->gDB->quote($field_id, "integer");
+
+		$res = $this->gDB->query($select);
+		while($rec = $this->gDB->fetchAssoc($res)) {
+			$ret[] = array("user_id" => $rec["usr_id"]
+						, "value" => $rec["value"]);
+			ilCronManager::ping($this->getId());
```

Comment:
Weil wir mal gesagt haben, immer wenn irgendwo was schief gehen kann dann ein ping. finde in so einer schleife kann dann schonmal was schief gehen.